src/main/java/seedu/duke/Parser.java

```diff
@@ -0,0 +1,115 @@
+package seedu.duke;
+
+/**
+ * Parser processes each command from Ui and executes program accordingly.
+ */
+public class Parser {
+    private TaskList taskLists;
+
+    /**
+     * Initializes an instance of a Parser.
+     *
+     * @param taskLists List of Tasks stored in a TaskList class.
+     */
+    public Parser(TaskList taskLists) {
+        this.taskLists = taskLists;
+    }
+
+    /**
+     * Checks for invalid inputs.
+     *
+     * @param input String that is input by user.
+     * @throws DukeException if input does not contain any of the keywords in the method.
+     */
+    public void checkForInvalidInput(String input) throws DukeException {
+        if (!input.contains("todo")) {
+            if (!input.contains("deadline")) {
+                if (!input.contains("event")) {
+                    if (!input.contains("done")) {
+                        if (!input.contains("bye")) {
+                            if (!input.contains("list")) {
+                                if (!input.contains("delete")) {
+                                    if (!input.contains("find")) {
+                                        Ui.printLines();
+                                        throw new DukeException("I don't know what that means! Try again.\n"
+                                                + "\n-----------------------------------------------\n");
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Processes input by user and execute accordingly.
+     *
+     * @param input String input by user.
+     */
+    public void read(String input) {
```

Comment:
Maybe a better name can be used for this method and the method below, for e.g. readCliInput vs readGuiInput